src/yarp.c

```diff
@@ -9267,7 +9267,7 @@ parse_expression_prefix(yp_parser_t *parser, yp_binding_power_t binding_power) {
         yp_binding_power_t binding_power = yp_binding_powers[parser->current.type].left;
 
         if (binding_power == YP_BINDING_POWER_UNSET || binding_power >= YP_BINDING_POWER_RANGE) {
-          arguments = yp_arguments_node_create(parser);
+          arguments = YP_NODE_DOWNCAST(yp_arguments_node_create(parser));
```

Comment:
Can you change the signature of `parse_arguments` to instead accept a `yp_arguments_node_t *`?